php/src/Operation.cpp

```diff
@@ -713,12 +709,11 @@ ZEND_FUNCTION(IcePHP_defineOperation)
 
     if (zend_parse_parameters(
             ZEND_NUM_ARGS(),
-            const_cast<char*>("osllla!a!a!a!"),
+            const_cast<char*>("oslla!a!a!a!"),
```

Comment:
This is black magic to me.
I removed a single `l` from the list, because it 'felt right'.
But I honestly have no idea what this is, so to anyone familiar, please double check this.

Reply:
https://github.com/php/php-src/blob/ef4b2fc283ddaf9bd692015f1db6dad52171c3ce/README.PARAMETER_PARSING_API#L43

Reply:
Does this format string line up correctly with the args we pass in?

Reply:
`o s l l a! a! a! a!`

I feel pretty confident about the last 4:
`a!`: `&inParams`   (array)
`a!`: `&outParams`   (array)
`a!`: `&returnType`   (array? but why is this an array?)
`a!`: `&exceptions`   (array)


But, the first 5 I'm less confident about. The only way I can get the numbers to add up is if `s` (for string) takes two arguments, a string, and a length. Then you get:

`o`: `&cls`  (object)
`s`: `&name, &nameLen`   (string)
`l`: `&mode`   (long?)
`l`: `&format`   (long?)

Otherwise, if `nameLen` should get it's own field, then it seems like we're missing one...
But this was already true before my PR.

Reply:
Okay, no more guessing. This is indeed the case:
```
 s  - string (with possible null bytes) and its length (char*, size_t)
```
https://github.com/php/php-src/blob/ef4b2fc283ddaf9bd692015f1db6dad52171c3ce/README.PARAMETER_PARSING_API#L68C1-L68C71

Reply:
And I guess `returnType` is an optional array because it could contain `0` or `1` elements?
(`0` meaning `void`)

Not clear why that's preferable to an optional object instead `O!`, but it doesn't really matter.